src/Module.Server/Rewards/CrpgRewardServer.cs

```diff
@@ -201,8 +201,16 @@ public override void OnRemoveBehavior()
             {
                 bool isValorousPlayer = valorousPlayerIds.Contains(playerId);
                 int compensationForCrpgUser = compensationByCrpgUserId.TryGetValue(crpgUserId, out int compensation) ? compensation : 0;
-                SetRewardForConnectedPlayer(userUpdate, crpgPeer, durationRewarded, compensationForCrpgUser, isValorousPlayer,
-                    defenderMultiplierGain, attackerMultiplierGain, constantMultiplier);
+                if (missionPeer.Team?.Side != BattleSideEnum.None)
+                {
+                    SetRewardForConnectedPlayer(userUpdate, crpgPeer, durationRewarded, compensationForCrpgUser, isValorousPlayer,
+                        defenderMultiplierGain, attackerMultiplierGain, constantMultiplier);
+                }
+                else
+                {
+                    SetRewardForConnectedPlayer(userUpdate, crpgPeer, 0f, compensationForCrpgUser, isValorousPlayer,
+                        defenderMultiplierGain, attackerMultiplierGain, constantMultiplier);
+                }

```

Comment:
What happens if I play a round and join spectator before its end?

Reply:
you will get 0 rewards

Reply:
you think it should give rewards?

Reply:
ok this would be more logic and i don't know if it's worth it, but what about another dictionary that saves the played duration on the OnAgentTeamChanged call and rewards the duration played?

Reply:
Actually I'm not sure I understood the problem. `LastSpawnInfo` should be null if they didn't spawn. Are you trying to prevent players joining spectator at the start of the round and going back to their team at the end of the round?

Reply:
For battle it's not a big deal.
The bigger problem is Conquest. When people spawn in and go to spectator immediatly, they still earn the rewards while staying in spectator. Effectively leeching.

Reply:
In that case I would just set their LastSpawnInfo to null. Not sure where that's the best to do that

Reply:
Problem is then the compensation.
If the people that did the teamhit just join spectator then the teamhitted mate gets gold, but the teamhitter no punishment.
Especially dangerous on DTV as it's possible to actually cheat gold that way because of the healing in that mode.

Reply:
Alright in that case I would just refactor this code to emphasize the difference between the 2 branches. Something like
```suggestion
                float durationRewardedForPlayer = missionPeer.Team?.Side != BattleSideEnum.None
                    ? durationRewarded
                    : 0f;
                SetRewardForConnectedPlayer(userUpdate, crpgPeer, durationRewardedForPlayer, compensationForCrpgUser, isValorousPlayer,
                    defenderMultiplierGain, attackerMultiplierGain, constantMultiplier);
```